test/Kongverge.IntegrationTests/ProgramSteps.cs

```diff
@@ -116,28 +117,40 @@ protected async Task OutputFolderContentsMatchesFolderContentsOf(string name)
             };
             var kongReader = new KongAdminReader(new KongAdminHttpClient(Options.Create(settings)));
             var kongConfiguration = await kongReader.GetConfiguration();
-            var availablePlugins = kongConfiguration.Plugins.Available.Where(x => x.Value).Select(x => x.Key).ToArray();
+            var availablePlugins = kongConfiguration.Plugins.Available.Where(x => x.Value).Select(x => x.Key).ToDictionary(x => x, x => new AsyncLazy<KongPluginSchema>(() => kongReader.GetPluginSchema(x)));
             var configReader = new ConfigFileReader();
-            var folderConfiguration = await configReader.ReadConfiguration(folder, availablePlugins);
-            var outputConfiguration = await configReader.ReadConfiguration(OutputFolder, availablePlugins);
+            var folderConfiguration = await configReader.ReadConfiguration(folder, availablePlugins.Keys);
+            var outputConfiguration = await configReader.ReadConfiguration(OutputFolder, availablePlugins.Keys);
 
             folderConfiguration.GlobalConfig.Plugins.Should().NotBeEmpty();
             folderConfiguration.Services.Count.Should().Be(3);
 
+            await FillInDefaults(folderConfiguration.GlobalConfig.Plugins, availablePlugins);
+
             outputConfiguration.GlobalConfig.Plugins.Should().BeEquivalentTo(folderConfiguration.GlobalConfig.Plugins);
             outputConfiguration.Services.Should().BeEquivalentTo(folderConfiguration.Services);
             foreach (var outputService in outputConfiguration.Services)
             {
                 var folderService = folderConfiguration.Services.Single(x => x.Name == outputService.Name);
+                await FillInDefaults(folderService.Plugins, availablePlugins);
                 outputService.Plugins.Should().BeEquivalentTo(folderService.Plugins);
                 outputService.Routes.Should().BeEquivalentTo(folderService.Routes);
                 foreach (var outputServiceRoute in outputService.Routes)
                 {
                     var folderServiceRoute = folderService.Routes.Single(x => x.Equals(outputServiceRoute));
+                    await FillInDefaults(folderServiceRoute.Plugins, availablePlugins);
                     outputServiceRoute.Plugins.Should().BeEquivalentTo(folderServiceRoute.Plugins);
                 }
             }
         }
+
+        private static async Task FillInDefaults(IEnumerable<KongPlugin> plugins, IDictionary<string, AsyncLazy<KongPluginSchema>> availablePlugins)
+        {
+            foreach (var globalConfigPlugin in plugins)
```

Comment:
The name globalConfigPlugin is misleading isn't it?
This function is applied to all objects with plugins.

Reply:
Ah, yes, that was leftover from extracting a method from where it was using just the global plugins. Will fix.